ChallengesWithTestsMarkVII/ChallengesSet03.cs

```diff
@@ -7,47 +7,152 @@ public class ChallengesSet03
+
         }
 
         public bool IsSumOfOddsOdd(IEnumerable<int> numbers)
         {
+
+            if (numbers == null)
+            {
+                return false;
+            }
+
+            int sum = 0;
+
+            foreach (int number in numbers)
+            {
+                if (number % 2 != 0)
+                {
+                    sum += number;
+                }
+            }
+
+            return (sum % 2 != 0);
+
             throw new NotImplementedException();
+
         }
 
         public bool PasswordContainsUpperLowerAndNumber(string password)
         {
+
+            bool containsUpper = false;
+            bool containsLower = false;
+            bool containsNumber = false;
+
+            foreach (char c in password)
+            {
+                if (Char.IsUpper(c))
+                {
+                    containsUpper = true;
+                }
+                else if (Char.IsLower(c))
+                {
+                    containsLower = true;
+                }
+                else if (Char.IsNumber(c))
+                {
+                    containsNumber = true;
+                }
+            }
+
+            return (containsUpper && containsLower && containsNumber);
+
             throw new NotImplementedException();
+
         }
 
         public char GetFirstLetterOfString(string val)
         {
+
+            return val[0];
+
             throw new NotImplementedException();
+
         }
 
         public char GetLastLetterOfString(string val)
         {
+
+            return val[val.Length-1]; ;
+
             throw new NotImplementedException();
+
         }
 
         public decimal Divide(decimal dividend, decimal divisor)
         {
+
+            if (divisor == 0)
+            {
+                return 0;
+            }
+            else
+            {
+                return dividend/divisor;
+            }
+
             throw new NotImplementedException();
+
         }
 
         public int LastMinusFirst(int[] nums)
         {
+
+            return nums[nums.Length-1] - nums[0];
+
             throw new NotImplementedException();
+
         }
 
         public int[] GetOddsBelow100()
         {
-            throw new NotImplementedException();
+
+            int maximum = 100;
+            int[] oddsBelowMaximum = new int[maximum/2];
+            int counter = 0;
+            /*
+            for (int i = 0; i < maximum; i++)
+            {
+                if (i%2 != 0)
+                {
+                    oddsBelow100[counter] = i;
+                    counter += 1;
+                }
+            }
+            */
+
+            for (int i = 1; i < maximum; i += 2)
+            {
+                oddsBelowMaximum[counter] = i;
+                counter += 1;
+            }
+
+            return oddsBelowMaximum;
         }
 
-        public void ChangeAllElementsToUppercase(string[] words)
+        public string[] ChangeAllElementsToUppercase(string[] words)
```

Comment:
This is OK, but not necessary.  The method is intended to modify what it was given, and, since arrays are reference types, the caller will have their variable modified "also".